action/protocol/execution/evm/evmstatedbadapter.go

```diff
@@ -406,6 +426,25 @@ func (stateDB *StateDBAdapter) Suicide(evmAddr common.Address) bool {
 		stateDB.logError(err)
 		return false
 	}
+	// before calling Suicide, EVM will transfer the contract's balance to beneficiary
+	// need to create a transaction log on successful suicide
+	if stateDB.lastAddBalanceAmount.Cmp(actBalance) == 0 {
+		if len(stateDB.lastAddBalanceAmount.Bytes()) > 0 {
```

Comment:
ignore efficiency and use `stateDB.lastAddBalanceAmount.Cmp(big.NewInt(0)) > 0`

Reply:
updated